### PR TITLE
[GNA] Fixed issue for concat connection to memory layer

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
- Fixed case when the model includes Memory input directly connected to Concat layer
- Added the new tests

### Tickets:
 - 56722

### CODE REVIEW

File: inference-engine/tests/functional/plugin/gna/pass_tests/concat_memory.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2018-2021 Intel Corporation
```
Comment (elilobanova) [inline inference-engine/tests/functional/plugin/gna/pass_tests/concat_memory.cpp:1]: I'm not sure if this test is related to pass tests. It doesn't test any passes but reservation of memory for layers.

**Review by nkogteva — APPROVED** [review]
LGTM

File: inference-engine/tests/functional/plugin/gna/pass_tests/concat_memory.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+protected:
+    void SetUp() override {
+        int axis;
+        std::vector<size_t> input_shape;
+        InferenceEngine::Precision net_precision;
+        std::tie(axis, input_shape, net_precision, targetDevice) = this->GetParam();
+        auto ng_prc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(net_precision);
+        auto input = ngraph::builder::makeParams(ng_prc, { input_shape });
+
+        auto variable = std::make_shared<ngraph::Variable>(ngraph::VariableInfo{ngraph::PartialShape::dynamic(), ngraph::element::dynamic, "v0"});
```
[the remark this answers is not in the document]
Comment (mryzhov) [reply inference-engine/tests/functional/plugin/gna/pass_tests/concat_memory.cpp]: fixed

File: inference-engine/tests/functional/plugin/gna/pass_tests/concat_memory.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        auto mem_i = std::make_shared<ngraph::opset8::Constant>(ng_prc, input_shape);
+        auto mem_r = std::make_shared<ngraph::opset8::ReadValue>(mem_i, variable);
+
+        ngraph::OutputVector concat_input;
+        concat_input.push_back(mem_r);
+        concat_input.push_back(input.at(0));
+        auto concat = std::make_shared<ngraph::opset8::Concat>(concat_input, axis);
+
+        auto mem_w = std::make_shared<ngraph::opset7::Assign>(input.at(0), variable);
```
[the remark this answers is not in the document]
Comment (mryzhov) [reply inference-engine/tests/functional/plugin/gna/pass_tests/concat_memory.cpp]: fixed

File: inference-engine/tests/functional/plugin/gna/pass_tests/concat_memory.cpp (hunk 1 above)
Comment (mryzhov) [reply inference-engine/tests/functional/plugin/gna/pass_tests/concat_memory.cpp]: moved the tests to the shared scope

**Review by elilobanova — APPROVED** [review]

**Pull request merged**